pytorch_lightning/plugins/environments/cluster_environment.py

```diff
@@ -40,6 +40,11 @@ def main_address(self) -> str:
     def main_port(self) -> int:
         """An open and configured port in the main node through which all processes communicate."""
 
+    @staticmethod
+    @abstractmethod
+    def detect() -> bool:
+        """Detects the environment settings corresponding to this cluster and returns ``True`` if they match."""
```

Comment:
total n00b question: any idea how to make adding a new abstractmethod not a breaking change? i imagine this will repeatedly come up

Reply:
one could avoid making it abstract now and remember to do it once the deprecation ends. not sure if there is a better way

Reply:
We could have a soft detect for custom cluster env. Return an Optional[bool] and have the default implementation passing. We could detect it hasn't been implemented and add a warning this function would be moved to abstract later on.